### ISSUE TITLE
Add support for Tomcat 10.0 and 10.1 

### ISSUE THREAD

**Issue by tfr42** [deegree/deegree3#1401]
To support Servlet API version 5.0 (TC 10.0) and 6.0 (TC 10.1) it is required to change the `javax.servlet.*` imports in the code by `jakarta.servlet.*`.

**Comment by tfr42** [comment]
The [Tomcat Migration Guide](https://tomcat.apache.org/migration-10.1.html) describes the requirements which need to be met. There is the [Tomcat Migration Tool for Jakarta EE](https://tomcat.apache.org/download-migration.cgi) to support the migration of an existing WAR file. How to run the tool is document in [here](https://github.com/apache/tomcat-jakartaee-migration).

**Comment by tfr42** [comment]
Steps for migrating to TC 10.x using Apache Maven 3.9+ with OpenJDK 17:

- update Maven plugins to run under Java 17
- update project dependencies that allow successfull compile and test execution under Java 17
- build WAR file with `mvn clean install`
- follow Tomcat Migration Guide and run Migration Tool against deegree WAR file
- deploy WAR file to docker container running Tomcat 10.x with JDK 17: `docker run -it --rm -v "$PWD"/deegree-services/deegree-webservices/target/deegree-webservices-${DEEGREE_VERSION}.war:/usr/local/tomcat/webapps/deegree-webservices.war -v "$HOME"/.deegree:/root/.deegree -p 8080:8080 tomcat:10.1.12-jdk17`

**Comment by tfr42** [comment]
A docker image with deegree webservices 3.5.2 on Tomcat 10.1 with JDK 11 is available for testing on hub.docker.com:
`docker pull tfr42/deegree:3.5-jdk17-tomcat10`

**Comment by julianzz98** [comment]
The docker image was successfully tested locally. 

The tests included:

- Basic functionality of WFS & WMS
- Validation of all examples via the console
- Database connection + data showcase via WFS

**Comment by tfr42** [comment]
Source code migration to `jakarta.*` with:
```
 mvn -U org.openrewrite.maven:rewrite-maven-plugin:run \
    -Drewrite.recipeArtifactCoordinates=org.openrewrite.recipe:rewrite-migrate-java:LATEST \
    -Drewrite.activeRecipes=org.openrewrite.java.migrate.jakarta.JavaxMigrationToJakarta
```

**Comment by tfr42** [comment]
The PR #1627 contains the upgrade to Servlet API 6.0 and Faces 4.0 as well as upgrade of #1625.

**Comment by tfr42** [comment]
To use JSF with CDI on Tomcat 10.1 it is required to enable CDI, see https://tomcat.apache.org/tomcat-10.1-doc/cdi.html for more information.

**Comment by tfr42** [comment]
The support of Servlet API version 5.0 (TC 10.0) and 6.0 (TC 10.1) was fully implemented with version 3.6 of deegree webservices, see https://github.com/deegree/deegree3/wiki/End-of-Life-and-Support-Matrix for more information.
